### PR TITLE
IGNITE-19670: Improve CatalogService test coverage.

### PR DESCRIPTION
**Pull request by lowka**
https://issues.apache.org/jira/browse/IGNITE-19670

---
Thank you for submitting the pull request.

To streamline the review process of the patch and ensure better code quality
we ask both an author and a reviewer to verify the following:

### The Review Checklist
- [ ] **Formal criteria:** TC status, codestyle, mandatory documentation. Also make sure to complete the following:  
\- There is a single JIRA ticket related to the pull request.  
\- The web-link to the pull request is attached to the JIRA ticket.  
\- The JIRA ticket has the Patch Available state.  
\- The description of the JIRA ticket explains WHAT was made, WHY and HOW.  
\- The pull request title is treated as the final commit message. The following pattern must be used: IGNITE-XXXX Change summary where XXXX - number of JIRA issue.
- [ ] **Design:** new code conforms with the design principles of the components it is added to.
- [ ] **Patch quality:** patch cannot be split into smaller pieces, its size must be reasonable.
- [ ] **Code quality:** code is clean and readable, necessary developer documentation is added if needed.
- [ ] **Tests code quality:** test set covers positive/negative scenarios, happy/edge cases. Tests are effective in terms of execution time and resources.

### Notes
- [Apache Ignite Coding Guidelines](https://cwiki.apache.org/confluence/display/IGNITE/Java+Code+Style+Guide)

### CODE REVIEW

**Review by zstan — APPROVED** [review]

File: modules/catalog/src/test/java/org/apache/ignite/internal/catalog/CatalogTableTest.java [hunk 1]
```diff
@@ -0,0 +1,1168 @@
...
+        assertThat(changeColumn(TABLE_NAME, "VAL", null, null, () -> constant(null)),
+                willCompleteSuccessfully());
+        assertNull(manager.schema(schemaVer + 1));
+
+        // NULL -> 1 : Ok.
+        assertThat(changeColumn(TABLE_NAME, "VAL", null, null, () -> constant(1)),
+                willCompleteSuccessfully());
+        assertNotNull(manager.schema(++schemaVer));
+
+        // 1 -> 1 : No-op.
```
Comment (AMashenkov) [inline modules/catalog/src/test/java/org/apache/ignite/internal/catalog/CatalogTableTest.java:684]: Should we check a functional default as well?

File: modules/catalog/src/test/java/org/apache/ignite/internal/catalog/CatalogZoneTest.java [hunk 2]
```diff
@@ -0,0 +1,464 @@
...
+            CatalogCommand dropCommand = DropZoneCommand.builder()
+                    .zoneName(initialDefaultZone.name())
+                    .build();
+
+            assertThat(manager.execute(dropCommand), willCompleteSuccessfully());
+        }
+    }
+
+    @Test
+    public void testDropZone() {
```
Comment (AMashenkov) [inline modules/catalog/src/test/java/org/apache/ignite/internal/catalog/CatalogZoneTest.java:175]: Let's also check that non-empty zone can't be dropped.

File: modules/catalog/src/test/java/org/apache/ignite/internal/catalog/CatalogZoneTest.java [hunk 3]
```diff
@@ -0,0 +1,464 @@
...
+        assertEquals(2, zone.replicas());
+        assertEquals(INFINITE_TIMER_VALUE, zone.dataNodesAutoAdjust());
+        assertEquals(3, zone.dataNodesAutoAdjustScaleUp());
+        assertEquals(4, zone.dataNodesAutoAdjustScaleDown());
+        assertEquals("newExpression", zone.filter());
+        assertEquals("test_profile", zone.storageProfiles().profiles().get(0).storageProfile());
+    }
+
+    @Test
+    public void testCreateZoneWithSameName() {
```
Comment (AMashenkov) [inline modules/catalog/src/test/java/org/apache/ignite/internal/catalog/CatalogZoneTest.java:374]: Let's also check if zone can be created after renaming/dropping the previous one with the same name.

**Review by AMashenkov — APPROVED** [review]
Feel free to create a ticket to extend test coverage, or add the tests within the current PR.

File: modules/catalog/src/test/java/org/apache/ignite/internal/catalog/CatalogZoneTest.java (hunk 3 above)
Comment (lowka) [reply modules/catalog/src/test/java/org/apache/ignite/internal/catalog/CatalogZoneTest.java:374]: Thanks. Added

File: modules/catalog/src/test/java/org/apache/ignite/internal/catalog/CatalogZoneTest.java (hunk 2 above)
Comment (lowka) [reply modules/catalog/src/test/java/org/apache/ignite/internal/catalog/CatalogZoneTest.java:175]: Such test case already exists in `DropZoneCommandValidationTest::rejectToDropZoneWithTable`

File: modules/catalog/src/test/java/org/apache/ignite/internal/catalog/CatalogTableTest.java (hunk 1 above)
Comment (lowka) [reply modules/catalog/src/test/java/org/apache/ignite/internal/catalog/CatalogTableTest.java:684]: Already covered under `AlterTableAlterColumnCommandValidationTest::cannotAddColumnWithFunctionalDefault`

**Pull request merged**